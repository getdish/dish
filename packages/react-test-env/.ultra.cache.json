{"files":{"jsdom-register.js":"1f2b9eb3ee0854a904d068ef38c188062c9cfd1c","package.json":"aa6fb98cd089c5f336c474b739cd8be471d39698","src/index.ts":"0d0e6d1839ab6865935dada399a3ab92cda38fd7","tsconfig.json":"4aeadf28b0899fa5e9af63cfdc361c39857eccf9"},"deps":{"@dish/esdx":1614675133074.2612,"@dish/helpers":1614675159800.7415}}