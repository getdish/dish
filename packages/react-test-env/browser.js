require('@dish/helpers-node').emulateBrowserEnv()
