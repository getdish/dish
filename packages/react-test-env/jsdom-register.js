const { JSDOM } = require('jsdom')

const url = 'http://d1sh_hasura_live.com:19006/'

// fake a browser!
const jsdom = new JSDOM(``, {
  pretendToBeVisual: true,
  url: url,
  referrer: url,
  contentType: 'text/html',
})

global['window'] = jsdom.window
Object.keys(jsdom.window).forEach((key) => {
  if (typeof global[key] === 'undefined') {
    global[key] = jsdom.window[key]
  }
})
