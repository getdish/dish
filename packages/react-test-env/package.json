{
  "name": "@dish/react-test-env",
  "version": "0.0.9",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "source": "src/index.ts",
  "main": "dist/cjs",
  "module": "dist/esm",
  "types": "types/index.d.ts",
  "scripts": {
    "build": "tamagui-build",
    "watch": "tamagui-build --watch"
  },
  "dependencies": {
    "@testing-library/react": "13.0.0-alpha.4",
    "debug-http": "^1.1.0",
    "happy-dom": "^2.24.4",
    "mutationobserver-polyfill": "^1.3.0",
    "react": "*",
    "react-dom": "*",
    "react-test-renderer": "18.2.0"
  },
  "devDependencies": {
    "@tamagui/build": "^1.6.3",
    "@types/react": "*",
    "@types/react-dom": "*",
    "@types/react-native": "*",
    "@types/react-test-renderer": "*"
  }
}
