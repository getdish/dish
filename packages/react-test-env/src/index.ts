import 'mutationobserver-polyfill'
// import { LocalStorage } from 'node-localstorage'
import React from 'react'
import ReactDOM from 'react-dom'

export { render, cleanup, fireEvent, waitFor, screen } from '@testing-library/react'

export { default as TestRenderer, act } from 'react-test-renderer'

// globalThis.localStorage = new LocalStorage('./tmp')

globalThis['React'] = React
globalThis['ReactDOM'] = ReactDOM

if (process.env.DEBUG) {
  const debugHttp = require('debug-http')
  debugHttp()
}
