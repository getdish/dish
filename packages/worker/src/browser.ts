export async function fetchBrowserJSON(url: string) {
  const res = await fetchBrowser(url, {
    'content-type': 'application/json',
    parse: 'json',
  })
  return await res.json()
}

export async function fetchBrowserHyperscript(url: string, selector: string) {
  const res = await fetchBrowser(url, {
    'content-type': 'application/json',
    parse: 'hyperscript',
    selector,
  })
  return await res.json()
}

export async function fetchBrowserHTML(url: string) {
  const res = await fetchBrowser(url)
  return await res.text()
}

const regions = ['atl', 'sea', 'lax', 'dfw', 'ord', 'iad', 'sjc', 'vin', 'yyz']
async function fetchBrowser(url: string, headers = {}, retry = 3) {
  let i = 0
  while (i < retry) {
    i++
    try {
      const region = regions[Math.round(Math.random() * regions.length)]
      return await fetch('https://dish-proxy.fly.dev', {
        headers: {
          url,
          'fly-prefer-region': region,
          ...headers,
        },
      })
    } catch (err) {
      console.error(`Error ${err.message}, retries left ${retry - i}`)
    }
  }
  throw new Error(`Failed to fetch`)
}
