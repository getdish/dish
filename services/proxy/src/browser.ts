import { Page, WebKitBrowser, webkit } from 'playwright'

if (!process.env.LUMINATI_PROXY_HOST) {
  throw new Error(`No proxy`)
}

let browser: WebKitBrowser | null = null
let browserHasLoadedOriginOnce = false
let page: Page | null = null
let tries = 0

const baseProxy = {
  server: `${process.env.LUMINATI_PROXY_HOST}:${process.env.LUMINATI_PROXY_PORT}`,
  username: process.env.LUMINATI_PROXY_RESIDENTIAL_USER,
  password: process.env.LUMINATI_PROXY_RESIDENTIAL_USER,
}

let activeProxy: typeof baseProxy | null = baseProxy

const awsProxies = {
  'https://ubereats.com/': process.env.UBEREATS_PROXY,
  'https://yelp.com/': process.env.YELP_AWS_PROXY,
  'https://infatuated.com/': process.env.INFATUATED_PROXY,
  'https://michelin.com/': process.env.MICHELIN_PROXY,
  'https://tripadvisor.com/': process.env.TRIPADVISOR_PROXY,
  'https://google.com/': process.env.GOOGLE_AWS_PROXY,
  'https://doordash_graphql.com/': process.env.DOORDASH_GRAPHQL_AWS_PROXY,
  'https://grubhub.com/': process.env.GRUBHUB_AWS_PROXY,
}

export async function ensurePage(forceRefresh = false) {
  if (!browser || forceRefresh) {
    tries++
    if (browser) {
      console.log('Force refresh from failed proxy, opening new browser')
      await browser.close()
    }
    if (tries > 1) {
      activeProxy = null
    }
    console.log('Starting browser, proxies:', activeProxy, awsProxies)
    browserHasLoadedOriginOnce = false
    browser = await webkit.launch({
      headless: true,
      ...(activeProxy && {
        proxy: activeProxy,
      }),
    })
    console.log('Browser version', browser.version())
    page = await browser.newPage()
  }
  return page!
}

export async function fetchBrowser(
  uriBase: string,
  type: 'html' | 'json' | 'hyperscript',
  selector?: string,
  maxTries = 3
) {
  let uri = uriBase
  if (!activeProxy) {
    uri = uri.replace('//www.', '//')
    for (const key in awsProxies) {
      if (uri.startsWith(key)) {
        console.log('Using aws proxy', key)
        uri = uri.replace(key, awsProxies[key])
      }
    }
  }
  console.log('fetchBrowser', uri)
  try {
    if (type === 'hyperscript') {
      return await fetchBrowserHyperscript(uri, selector ?? '')
    }
    if (type === 'html') {
      return await fetchBrowserHTML(uri)
    }
    return await fetchBrowserJSON(uri)
  } catch (err) {
    if (maxTries > 0) {
      console.log(`Trying again (${maxTries} tries left)`)
      await ensurePage(true)
      return await fetchBrowser(uri, type, selector, maxTries - 1)
    } else {
      console.error(`Failed: ${err.message}`)
    }
  }
}

export async function fetchBrowserJSON(uri: string, retry = 0) {
  const page = await ensurePage()
  const url = new URL(uri)
  try {
    if (!browserHasLoadedOriginOnce) {
      console.log('loading origin once first')
      await page.goto(url.origin)
      browserHasLoadedOriginOnce = true
    }
    if (retry == 0) {
      // first attempt, lets try fetch inline
      try {
        console.log('inline fetch', uri)
        return await timeout(
          page.evaluate((uri: string) => {
            return fetch(uri, {
              headers: {
                'content-type': 'application/json',
                accept: 'application/json',
                'Accept-Encoding': 'br;q=1.0, gzip;q=0.8, *;q=0.1',
              },
            }).then((res) => res.json())
          }, uri),
          5_000
        )
      } catch (err) {
        console.error(`Error inline fetching: ${err.message}`)
        // allow to continue
      }
    }
    console.log('goto url', uri)
    await page.goto(uri, {
      waitUntil: 'domcontentloaded',
      timeout: 10_000,
    })
    console.log('get text content')
    const out = await page.textContent('body', {
      timeout: 10_000,
    })
    if (out) {
      console.log('got body', typeof out, out)
      return JSON.parse(out) as { [key: string]: any }
    }
    console.log('got no out')
  } catch (err) {
    console.error(`Error: ${err.message}`)
    if (retry < 2) {
      console.log('rety', retry)
      browserHasLoadedOriginOnce = false
      return await fetchBrowserJSON(uri, retry + 1)
    }
    throw err
  }
  return null
}

export async function fetchBrowserHTML(uri: string) {
  try {
    const page = await ensurePage()
    await page.goto(uri)
    const content = await page.content()
    console.log('returning html content', content.slice(0, 50) + '...')
    return content
  } catch (err) {
    console.error(`Error: ${err.message}`)
    throw err
  }
}

export async function fetchBrowserHyperscript(uri: string, selector: string) {
  const page = await ensurePage()
  await page.goto(uri)
  console.log('hyperscript selector', selector)
  const content = await page.textContent(selector)
  if (content) {
    console.log('got hyperscript, parsing..', content.slice(0, 50) + '...')
    return hyperDecode(content)
  }
  return null
}

function hyperDecode(jsonPayload: string) {
  return decode(
    jsonPayload.slice(LEFT.length, jsonPayload.length - RIGHT.length)
  )
}
const LEFT = '<!--'
const RIGHT = '-->'
const ENCODE = [
  ['&', '&amp;'],
  ['>', '&gt;'],
]
function decode(res) {
  const jsonPayload = ENCODE.reduceRight((str, coding) => {
    const [encodeChar, htmlEntity] = coding
    return str.replace(new RegExp(htmlEntity, 'g'), encodeChar)
  }, res)

  return JSON.parse(jsonPayload)
}

async function timeout<A extends Promise<any>>(
  promise: A,
  time = 0
): Promise<any> {
  const failed = Symbol()
  let tm
  // @ts-ignore
  const res = await Promise.race([
    promise,
    new Promise((res) => {
      tm = setTimeout(() => res(failed), time)
    }),
  ])
  if (res === failed) {
    throw new Error(`Timed out in ${time}ms`)
  }
  clearTimeout(tm)
  return res
}
