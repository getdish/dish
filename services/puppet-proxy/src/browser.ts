import { Page, WebKitBrowser, devices, webkit } from 'playwright-webkit'

if (!process.env.LUMINATI_PROXY_HOST) {
  throw new Error(`No proxy config`)
}

const awsProxies = {
  'https://ubereats.com/': process.env.UBEREATS_PROXY,
  'https://yelp.com/': process.env.YELP_AWS_PROXY,
  'https://m.yelp.com/': process.env.YELP_MOBILE_AWS_PROXY,
  'https://infatuated.com/': process.env.INFATUATED_PROXY,
  'https://michelin.com/': process.env.MICHELIN_PROXY,
  'https://tripadvisor.com/': process.env.TRIPADVISOR_PROXY,
  'https://google.com/': process.env.GOOGLE_AWS_PROXY,
  'https://doordash_graphql.com/': process.env.DOORDASH_GRAPHQL_AWS_PROXY,
  'https://grubhub.com/': process.env.GRUBHUB_AWS_PROXY,
}

const baseProxy = {
  server: `${process.env.LUMINATI_PROXY_HOST}:${process.env.LUMINATI_PROXY_PORT}`,
  username: process.env.LUMINATI_PROXY_RESIDENTIAL_USER,
  password: process.env.LUMINATI_PROXY_RESIDENTIAL_USER,
}

const instances: {
  [key: string]: {
    browser: WebKitBrowser | null
    page: Page | null
    tries: number
    proxy: null | string | typeof baseProxy
  }
} = {}

function getProxyURL(full: string, useAWS = false) {
  let uri = full
  const proxy = (() => {
    if (useAWS) {
      const proxyURI = uri.replace('//www.', '//')
      for (const key in awsProxies) {
        if (proxyURI.startsWith(key)) {
          console.log('Using aws proxy', key, uri)
          uri = proxyURI.replace(key, awsProxies[key])
          return null
        }
      }
      console.error(`No aws proxy found`)
      return null
    } else {
      return baseProxy
    }
  })()
  return {
    uri,
    isMobile: full.includes('//m.'),
    origin: new URL(uri).origin,
    proxy,
  }
}

type CreateBrowserOpts = { recreate?: boolean; useAWS?: boolean }

async function ensureInstance(
  fullURL: string,
  { recreate = false, useAWS = false }: CreateBrowserOpts = {}
) {
  try {
    const url = getProxyURL(fullURL, useAWS)
    const instance = instances[url.origin]
    if (!instance || recreate) {
      if (instance) {
        console.log('closing existing...')
        await instance.browser.close()
      }
      console.log('start browser with', { recreate, useAWS }, url)
      const browser = await webkit.launch({
        headless: true,
        ...(url.proxy && {
          proxy: url.proxy,
        }),
      })
      const contextConfig = url.isMobile
        ? {
            ...devices['iPhone 11 Pro'],
          }
        : {}
      console.log('using context', contextConfig)
      const context = await browser.newContext(contextConfig)
      const page = await context.newPage()
      if (process.env.NODE_ENV === 'production') {
        console.log('loading origin once first', url.origin)
        await page.goto(url.origin)
      }
      console.log('creating page')
      instances[url.origin] = {
        proxy: url.proxy,
        tries: instance?.tries ?? 0,
        browser,
        page,
      }
    }
    return instances[url.origin]!
  } catch (err) {
    console.log(`Error creating browser instance`, err.message, err.stack)
    throw err
  }
}

async function getBrowserAndURL(uri: string, retry = 0, userOpts?: CreateBrowserOpts) {
  const opts = {
    useAWS: retry > 1 ? true : false,
    recreate: retry < 1,
    ...userOpts,
  }
  console.log('using opts', opts)
  const instance = await ensureInstance(uri, opts)
  const url = getProxyURL(uri, opts.useAWS)
  return {
    instance,
    url,
  }
}

export async function fetchBrowser(
  uri: string,
  type: 'html' | 'json' | 'script-data',
  selectors?: string[] | null,
  retry = 3
) {
  try {
    if (type === 'script-data') {
      return await fetchBrowserScriptData(uri, selectors ?? [], retry)
    }
    if (type === 'html') {
      return await fetchBrowserHTML(uri, retry)
    }
    return await fetchBrowserJSON(uri, retry)
  } catch (err) {
    if (retry > 0) {
      console.log('error fetching', err)
      console.log(`Trying again (retry ${retry})`)
      return await fetchBrowser(uri, type, selectors ?? [], retry - 1)
    } else {
      console.error(`Failed: ${err.message}`)
    }
  }
}

export async function fetchBrowserJSON(uri: string, retry = 0) {
  const { instance, url } = await getBrowserAndURL(uri, retry)
  if (!instance || !instance.page) throw new Error(`No browser instance`)
  try {
    // all except last two attempts, do it inline for less resource usage
    if (retry > 1) {
      try {
        console.log('inline fetch', url)
        return await timeout(
          instance.page.evaluate((uri: string) => {
            return fetch(uri, {
              headers: {
                'content-type': 'application/json',
                accept: 'application/json',
                'Accept-Encoding': 'br;q=1.0, gzip;q=0.8, *;q=0.1',
              },
            }).then((res) => res.json())
          }, url.uri),
          10_000
        )
      } catch (err) {
        console.error(`Error inline fetching, will try navigating: ${err.message}`)
        // allow to continue
      }
    }
    console.log('goto url', url.uri)
    await instance.page.goto(url.uri, {
      waitUntil: 'domcontentloaded',
      timeout: 10_000,
    })
    console.log('get text content')
    const out = await instance.page.textContent('body', {
      timeout: 10_000,
    })
    if (out) {
      console.log('got body', typeof out, out.slice(0, 1000))
      return JSON.parse(out) as { [key: string]: any }
    }
    console.log('got no out')
  } catch (err) {
    console.error(`Error: ${err.message}`)
    if (retry < 2) {
      console.log('err retry', retry)
      return await fetchBrowserJSON(uri, retry + 1)
    }
    throw err
  }
  return null
}

export async function fetchBrowserHTML(uri: string, retry = 0) {
  try {
    const {
      instance: { page },
      url,
    } = await getBrowserAndURL(uri, retry)
    if (!page) throw new Error(`No browser instance`)
    console.log('fetch browser html', uri, retry)
    await page.goto(url.uri, {
      timeout: 60_000,
    })
    const content = await page.content()
    console.log('returning html content', content.slice(0, 50) + '...')
    return content
  } catch (err) {
    console.error(`Error: ${err.message}`)
    throw err
  }
}

export async function fetchBrowserScriptData(uri: string, selectors: string[], retry = 0) {
  const {
    instance: { page },
    url,
  } = await getBrowserAndURL(uri, retry)
  if (!page) throw new Error(`No browser instance`)
  console.log('fetch browser hyperscript', uri, retry)
  await page.goto(url.uri, {
    timeout: 60_000,
  })

  let results: any[] = []

  console.log('script selectors', selectors)
  if (!selectors) {
    console.log('error no selectors')
    return
  }

  for (const selector of selectors) {
    const handles = await page.$$(selector)
    if (handles) {
      results.push(
        await Promise.all(
          handles.map(async (handle) => {
            try {
              const isHyperscript = !!(await handle.getAttribute('data-hypernova-key'))
              if (isHyperscript) {
                const textContent = await handle.textContent()
                if (!textContent) {
                  return null
                }
                console.log('got hyperscript, parsing..', textContent.slice(0, 50) + '...')
                return hyperDecode(textContent)
              }
              const isLDJSON = (await handle.getAttribute('type')) == 'application/ld+json'
              if (isLDJSON) {
                const textContent = await handle.textContent()
                if (!textContent) {
                  return null
                }
                console.log('got ldjson')
                return JSON.parse(textContent)
              }
              return await handle.innerHTML()
            } catch (err) {
              console.log('error processing handle', err.message, err.stack)
              return null
            }
          })
        )
      )
    }
  }

  return results
}

function hyperDecode(jsonPayload: string) {
  return decode(jsonPayload.slice(LEFT.length, jsonPayload.length - RIGHT.length))
}
const LEFT = '<!--'
const RIGHT = '-->'
const ENCODE = [
  ['&', '&amp;'],
  ['>', '&gt;'],
]
function decode(res) {
  const jsonPayload = ENCODE.reduceRight((str, coding) => {
    const [encodeChar, htmlEntity] = coding
    return str.replace(new RegExp(htmlEntity, 'g'), encodeChar)
  }, res)

  return JSON.parse(jsonPayload)
}

async function timeout<A extends Promise<any>>(promise: A, time = 0): Promise<any> {
  const failed = Symbol()
  let tm
  // @ts-ignore
  const res = await Promise.race([
    promise,
    new Promise((res) => {
      tm = setTimeout(() => res(failed), time)
    }),
  ])
  if (res === failed) {
    throw new Error(`Timed out in ${time}ms`)
  }
  clearTimeout(tm)
  return res
}
