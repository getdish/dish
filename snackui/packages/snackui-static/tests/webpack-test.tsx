import '@dish/react-test-env/browser'
import '@expo/match-media'

import { TestRenderer, act, render } from '@dish/testy'
import React from 'react'

const app = require('./spec/out/out-webpack')

let context: any = {}

beforeAll(async () => {
  await Promise.all([
    Object.keys(app).map((key) => {
      return act(() => {
        const App = app[key]
        context[key.toLowerCase()] = {
          Element: App,
          renderer: TestRenderer.create(<App conditional={true} />),
          rendererFalse: TestRenderer.create(<App conditional={false} />),
        }
      })
    }),
  ])
})

// TODO fix testability of linear gradient
// test('extracts gradients', () => {
//   const { renderer } = context.testlineargradient
//   console.log(renderer.toJSON())
//   // console.log('out', style.backgroundColor, style.paddingRight)
//   t.assert(true)
// })

// test('extracts media queries', async () => {
//   const { TestMediaQuery } = context.app
//   const { style } = await getTestElement(TestMediaQuery)
//   // TODO not picking up media queries
//   console.log('out', style.backgroundColor, style.paddingRight)
//   expect(true)
// })

//
// test styles
//
// testStyles(test)

test('1. extracts to a div for simple views, flat transforms', () => {
  const { test1 } = context
  const out = test1.renderer.toJSON()!
  expect(out).toMatchSnapshot()
})

test('2. extracts className for complex views but keeps other props', () => {
  const { test2 } = context
  const out = test2.renderer.toJSON()!
  expect(out).toMatchSnapshot()
  const outFalse = test2.rendererFalse.toJSON()
  expect(outFalse).toMatchSnapshot()
})

test('3. places className correctly given a single spread', () => {
  const {
    test3: { Element },
  } = context
  const out = render(<Element />)
  const list = [...out.container.firstChild?.['classList']]
  expect(list).toMatchSnapshot()
})

test('4. leaves dynamic variables', () => {
  const {
    test4: { renderer, Element },
  } = context
  const out = render(<Element />)
  const firstChild = out.container.firstChild!
  const classList = [...firstChild['classList']]
  expect(classList).toMatchSnapshot()
  const r = renderer.toJSON()
  expect(r).toMatchSnapshot()
})

test('5. spread conditional', () => {
  const { test5 } = context
  const out = test5.renderer.toJSON()
  expect(out).toMatchSnapshot()
})

test('6. spread ternary', () => {
  const { test6 } = context
  expect(test6.renderer.toJSON()).toMatchSnapshot()
  expect(test6.rendererFalse.toJSON()).toMatchSnapshot()
})

test('7. ternary + data-is', () => {
  const { test7 } = context
  const out = test7.renderer.toJSON()
  expect(out).toMatchSnapshot()
})

test('8. styleExpansions', () => {
  const { test8 } = context
  const out = test8.renderer.toJSON()
  expect(out).toMatchSnapshot()
  // TODO test constant folding
})

test('9. combines with classname', () => {
  const { test9 } = context
  const out = test9.renderer.toJSON()
  expect(out).toMatchSnapshot()
})

test('10. extracts Text', () => {
  const { test10 } = context
  const out = test10.renderer.toJSON()
  expect(out).toMatchSnapshot()
})

test('11. combines everything', () => {
  const {
    test11: { Element },
  } = context
  const out = render(<Element conditional={false} />)
  const firstChild = out.container.firstChild!
  const classList = [...firstChild['classList']]
  expect(classList).toMatchSnapshot()
})

test('12. ternary multiple on same key', () => {
  const { test12 } = context
  expect(test12.renderer.toJSON()).toMatchSnapshot()
})

// test('13. text with complex conditional and local vars', () => {
//   const { test13 } = context
//   // console.log('test13', test13.renderer!.toTree())
//   t.is(1, 1)
// })

test('14. extracts psuedo styles and evaluates constants', () => {
  const { test14 } = context
  const out = test14.renderer.toJSON()
  expect(out).toMatchSnapshot()
})

test('15. extracts spacer (complex expansion)', () => {
  const { test15 } = context
  const out = test15.renderer.toJSON()
  expect(out).toMatchSnapshot()
})

test('16. deopt when spreading multiple', () => {
  const { test16 } = context
  const out = test16.renderer.toJSON()
  expect(out).toMatchSnapshot()
})
