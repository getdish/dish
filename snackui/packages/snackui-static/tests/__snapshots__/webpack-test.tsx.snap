// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. extracts to a div for simple views, flat transforms 1`] = `
<div
  className=" test1 _tr--hover-scalez2z _flSh-1 _flGr-1 _flDi-column _flBa-auto _di-flex _boSi-borderzbox _boSh-e8nksv _boToRiRa-100px _boToLeRa-100px _boBoRiRa-100px _boBoLeRa-100px _baCo-19sdxe _alIt-stretch"
>
  <span
    className=" _di-inline"
  >
    hello world
  </span>
</div>
`;

exports[`2. extracts className for complex views but keeps other props 1`] = `
Array [
  <div
    className="css-view-mht2lk _alIt-stretch _boSi-borderzbox _di-flex _flBa-auto _flDi-column _flSh-zz _ovX-hidden _ovY-hidden who _boSh-xgwarg _boBoLeRa-12px _boBoRiRa-12px _boToLeRa-12px _boToRiRa-12px _paBo-5px _paLe-5px _paRi-5px _paTo-5px r-alignItems-1oszu61 r-boxSizing-deolkf r-display-6koalj r-flexBasis-1mlwlqe r-flexDirection-eqz5dr r-flexShrink-1q142lx"
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,1.00)",
      }
    }
  />,
  <div
    className="css-view-mht2lk ease-in-out-top _paTo-2px _paBo-2px _flSh-zz _flDi-column _flBa-auto _di-flex _boSi-borderzbox _alIt-stretch _baCo-1d0hial _to-z14px _boSh-xgwarg _boBoLeRa-12px _boBoRiRa-12px _boToLeRa-12px _boToRiRa-12px _paLe-5px _paRi-5px r-alignItems-1oszu61 r-boxSizing-deolkf r-display-6koalj r-flexBasis-1mlwlqe r-flexDirection-eqz5dr r-flexShrink-1q142lx"
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,1.00)",
      }
    }
  >
    <span
      className=" _di-inline"
    >
      hello world
    </span>
  </div>,
]
`;

exports[`2. extracts className for complex views but keeps other props 2`] = `
Array [
  <div
    className="css-view-mht2lk _alIt-stretch _boSi-borderzbox _di-flex _flBa-auto _flDi-column _flSh-zz _ovX-hidden _ovY-hidden who _boSh-xgwarg _boBoLeRa-12px _boBoRiRa-12px _boToLeRa-12px _boToRiRa-12px _paBo-5px _paLe-5px _paRi-5px _paTo-5px r-alignItems-1oszu61 r-boxSizing-deolkf r-display-6koalj r-flexBasis-1mlwlqe r-flexDirection-eqz5dr r-flexShrink-1q142lx"
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,1.00)",
      }
    }
  />,
  <div
    className="css-view-mht2lk ease-in-out-top _to-0px _paTo-2px _paBo-2px _flSh-zz _flDi-column _flBa-auto _di-flex _boSi-borderzbox _baCo-rhzyip _alIt-stretch _boSh-xgwarg _boBoLeRa-12px _boBoRiRa-12px _boToLeRa-12px _boToRiRa-12px _paLe-5px _paRi-5px r-alignItems-1oszu61 r-boxSizing-deolkf r-display-6koalj r-flexBasis-1mlwlqe r-flexDirection-eqz5dr r-flexShrink-1q142lx"
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,1.00)",
      }
    }
  >
    <span
      className=" _di-inline"
    >
      hello world
    </span>
  </div>,
]
`;

exports[`3. places className correctly given a single spread 1`] = `
Array [
  "css-view-mht2lk",
  "_alIt-stretch",
  "_boSi-borderzbox",
  "_di-flex",
  "_flBa-auto",
  "_flDi-column",
  "_flSh-zz",
  "_ovX-hidden",
  "_ovY-hidden",
  "r-alignItems-1oszu61",
  "r-boxSizing-deolkf",
  "r-display-6koalj",
  "r-flexBasis-1mlwlqe",
  "r-flexDirection-eqz5dr",
  "r-flexShrink-1q142lx",
]
`;

exports[`4. leaves dynamic variables 1`] = `
Array [
  "css-view-mht2lk",
  "_he-200px",
  "_flSh-zz",
  "_flDi-column",
  "_flBa-auto",
  "_di-flex",
  "_boSi-borderzbox",
  "_alIt-stretch",
  "_ovY--hover-visible",
  "_ovX--hover-visible",
  "r-alignItems-1oszu61",
  "r-boxSizing-deolkf",
  "r-display-6koalj",
  "r-flexBasis-1mlwlqe",
  "r-flexDirection-eqz5dr",
  "r-flexShrink-1q142lx",
]
`;

exports[`4. leaves dynamic variables 2`] = `
<div
  className="css-view-mht2lk  _he-200px _flSh-zz _flDi-column _flBa-auto _di-flex _boSi-borderzbox _alIt-stretch _ovY--hover-visible _ovX--hover-visible r-alignItems-1oszu61 r-boxSizing-deolkf r-display-6koalj r-flexBasis-1mlwlqe r-flexDirection-eqz5dr r-flexShrink-1q142lx"
  style={
    Object {
      "width": "calc(100% + 20px)",
    }
  }
/>
`;

exports[`5. spread conditional 1`] = `
<div
  className="_baCo-183yn7 _alIt-stretch _boSi-borderzbox _di-flex _flBa-auto _flDi-column _flSh-zz _ovX-hidden _ovY-hidden"
>
  <div
    className="_baCo-183yn7 _alIt-stretch _boSi-borderzbox _di-flex _flBa-auto _flDi-column _flSh-zz _ovX-hidden _ovY-hidden hello-world"
  />
</div>
`;

exports[`6. spread ternary 1`] = `
<div
  className="_baCo-183yn7 _alIt-stretch _boSi-borderzbox _di-flex _flBa-auto _flDi-column _flSh-zz _ovX-hidden _ovY-hidden"
>
  <span
    className=" _di-inline"
  >
    hello world
  </span>
</div>
`;

exports[`6. spread ternary 2`] = `
<div
  className="_baCo-19sdxe _alIt-stretch _boSi-borderzbox _di-flex _flBa-auto _flDi-column _flSh-zz _ovX-hidden _ovY-hidden"
>
  <span
    className=" _di-inline"
  >
    hello world
  </span>
</div>
`;

exports[`7. ternary + data-is 1`] = `
<div
  className=" _wi-66pct _paRi-11px _paLe-11px _paBo-10px _miWi-500px _maWi-30pct _flSh-zz _flDi-column _flBa-auto _di-flex _boSi-borderzbox _alIt-stretch"
>
  <div
    className="_wi-10px _he-10px _alIt-stretch _boSi-borderzbox _di-flex _flBa-auto _flDi-column _flSh-zz"
  />
</div>
`;

exports[`8. styleExpansions 1`] = `
<div
  className="css-view-mht2lk  _to-0px _ri-0px _po-relative _le-0px _flSh-zz _flDi-column _flBa-auto _di-flex _boSi-borderzbox _bo-0px _alIt-stretch r-alignItems-1oszu61 r-boxSizing-deolkf r-display-6koalj r-flexBasis-1mlwlqe r-flexDirection-eqz5dr r-flexShrink-1q142lx r-position-u8s1d"
  style={
    Object {
      "bottom": "0px",
      "left": "0px",
      "right": "0px",
      "top": "0px",
    }
  }
>
  <div
    className=" _po-absolute _flSh-zz _flDi-column _flBa-auto _di-flex _boSi-borderzbox _baCo-19sdxe _alIt-stretch"
  >
    <div
      className="css-view-mht2lk  _to-0px _ri-0px _po-relative _le-0px _flSh-zz _flDi-column _flBa-auto _di-flex _boSi-borderzbox _bo-0px _baCo-1d0hial _alIt-stretch r-alignItems-1oszu61 r-boxSizing-deolkf r-display-6koalj r-flexBasis-1mlwlqe r-flexDirection-eqz5dr r-flexShrink-1q142lx r-position-u8s1d"
      style={
        Object {
          "bottom": "0px",
          "left": "0px",
          "right": "0px",
          "top": "0px",
        }
      }
    />
  </div>
</div>
`;

exports[`9. combines with classname 1`] = `
<div
  className=" _paTo-15px _paBo-15px _flSh-zz _flDi-column _flBa-auto _di-flex _boSi-borderzbox _alIt-stretch home-top-dish"
/>
`;

exports[`10. extracts Text 1`] = `
<span
  className=" _foSi-10px _di-inline"
>
  <div
    className="css-text-1u9zv9r  _usSe-text _di-inline r-display-1471scf r-userSelect-1xnzce8"
    style={
      Object {
        "selectable": true,
      }
    }
  >
    <span
      className="css-text-1u9zv9r css-textHasAncestor-16my406 _di-inline _foSi-15px _foWe-600 _col-18s66rc r-display-xoduu5"
    >
      hello
    </span>
  </div>
</span>
`;

exports[`11. combines everything 1`] = `
Array [
  "css-view-mht2lk",
  "_baCo-183yn7",
  "_alIt-center",
  "_miHe-10px",
  "_ovX-hidden",
  "_ovY-hidden",
  "_po-relative",
  "_boToCo-dic8vh",
  "_boRiCo-1n8flkp",
  "_boLeCo-wse84g",
  "_boBoCo-1qvh775",
  "_boBoSt-solid",
  "_boBoWi-1px",
  "_boLeSt-solid",
  "_boLeWi-1px",
  "_boRiSt-solid",
  "_boRiWi-1px",
  "_boToSt-solid",
  "_boToWi-1px",
  "_boSi-borderzbox",
  "_di-flex",
  "_flBa-auto",
  "_flDi-column",
  "_flSh-zz",
  "_he-0px",
  "_boToRiRa-0px",
  "_boToLeRa-0px",
  "_boBoRiRa-0px",
  "_boBoLeRa-0px",
  "r-alignItems-1oszu61",
  "r-boxSizing-deolkf",
  "r-display-6koalj",
  "r-flexBasis-1mlwlqe",
  "r-flexDirection-eqz5dr",
  "r-flexShrink-1q142lx",
]
`;

exports[`12. ternary multiple on same key 1`] = `
<div
  className="_tr- _op-1 _alIt-stretch _boSi-borderzbox _di-flex _flBa-auto _flDi-column _flSh-zz"
/>
`;

exports[`14. extracts psuedo styles and evaluates constants 1`] = `
<div
  className=" _baCo--active-183yn7 _baCo--hover-19sdxe _flSh-zz _flDi-column _flBa-auto _di-flex _boSi-borderzbox _alIt-stretch"
/>
`;

exports[`15. extracts spacer (complex expansion) 1`] = `
Array [
  <div
    className=" _miWi-12px _miHe-12px _flSh-zz _fl-zz"
  />,
  <div
    className=" _miWi-10px _miHe-10px _flSh-zz _fl-1"
  />,
]
`;

exports[`16. deopt when spreading multiple 1`] = `
<div
  className="css-view-mht2lk _baCo-19sdxe _alIt-stretch _boSi-borderzbox _di-flex _flBa-auto _flDi-column _flSh-zz _flGr-1 r-alignItems-1oszu61 r-boxSizing-deolkf r-display-6koalj r-flexBasis-1mlwlqe r-flexDirection-eqz5dr r-flexShrink-1q142lx"
/>
`;
